server/model/monitor.js

```diff
@@ -175,6 +177,44 @@ class Monitor extends BeanModel {
                     bean.ping = await ping(this.hostname);
                     bean.msg = ""
                     bean.status = UP;
+                } else if (this.type === "dns") {
+                    let startTime = dayjs().valueOf();
+
+                    var dnsRes = await dnsResolve(this.hostname, this.dns_resolve_server, this.dns_resolve_type);
+
+                    var dnsMessage = "";
+
+                    if (this.dns_resolve_type == 'A' || this.dns_resolve_type == 'AAAA' || this.dns_resolve_type == 'CNAME' || this.dns_resolve_type == 'PTR') {
+                        var dnsMessage = dnsRes[0];
+                    } else if (this.dns_resolve_type == 'CAA') {
+                        var dnsMessage = dnsRes[0].issue;
+                    } else if (this.dns_resolve_type == 'MX') {
+                        dnsRes.forEach(record => {
+                            dnsMessage += `Server: ${record.exchange} - Priority: ${record.priority} | `;
+                        });
+                        var dnsMessage = dnsMessage.slice(0, -2)
+                    } else if (this.dns_resolve_type == 'NS') { 
+                        dnsRes.forEach(record => {
+                            dnsMessage += `Server: ${record} | `;
+                        });
+                        var dnsMessage = dnsMessage.slice(0, -2)
+                    } else if (this.dns_resolve_type == 'SOA') {
+                        dnsMessage += `NS-Name: ${dnsRes.nsname} | Hostmaster: ${dnsRes.hostmaster} | Serial: ${dnsRes.serial} | Refresh: ${dnsRes.refresh} | Retry: ${dnsRes.retry} | Expire: ${dnsRes.expire} | MinTTL: ${dnsRes.minttl}`;
+                    } else if (this.dns_resolve_type == 'SRV') {
+                        dnsRes.forEach(record => {
+                            dnsMessage += `Name: ${record.name} | Port: ${record.port} | Priority: ${record.priority} | Weight: ${record.weight} | `;
+                        });
+                        var dnsMessage = dnsMessage.slice(0, -2)
+                    } else if (this.dns_resolve_type == 'TXT') {
+                        dnsRes.forEach(record => {
+                            dnsMessage += `Record: ${record} | `;
+                        });
+                        var dnsMessage = dnsMessage.slice(0, -2)
+                    }
+
+                    bean.msg = dnsMessage;
+                    bean.ping = dayjs().valueOf() - startTime;
```

Comment:
Suggest moving `bean.ping = ...` to line 184 to increase timing accuracy.

Also please fix eslint errors by using let instead of var and remove re-declaration.

Reply:
> Suggest moving `bean.ping = ...` to line 184 to increase timing accuracy.
> 
> Also please fix eslint errors by using let instead of var and remove re-declaration.

Thanks for the suggestions!
Sorry for the question because i've never really worked with eslint, how can i see them?

Reply:
Unfortunately we don't have a lint command setup yet, I'll work on it ASAP. For now the easiest way is still to install ESLint extension in your code editor (VS Code in my case).

Reply:
Thank you! I fixed the ESLint errors from my code.

Reply:
We added lint commands in this commit: https://github.com/louislam/uptime-kuma/commit/833b4733a8627e195384a631c9b7b55c7d3bcc24